### PR TITLE
Feat/fnm support

### LINKED ISSUE #847: Support FNM Node Version Manager in Node configuration provider
Support FNM Node Version Manager (https://github.com/Schniz/fnm) in Node configuration provider.

The change would be in this method:
https://github.com/Microsoft/vscode-node-debug/blob/master/src/node/extension/configurationProvider.ts#L222

The location for each `$VERSION` is here: `$HOME/.fnm/node-versions/$VERSION/installation/bin/node`

### PR DESCRIPTION
**Pull request by thales-maciel**
Implements support for [fnm node manager](https://github.com/Schniz/fnm)

This closes #847.

### CODE REVIEW

**Comment by thales-maciel** [comment]
@microsoft-github-policy-service agree

**Comment by thales-maciel** [comment]
I'm sorry about the commit spam, I couldn't run the tests locally for some reason.

**Comment by connor4312** [comment]
No worries. This week is release week, I will review this next week. Thanks for the PR! :)

File: src/targets/node/nvmResolver.ts [hunk 1]
```diff
@@ -87,6 +89,17 @@ export class NvmResolver implements INvmResolver {
       }
     }
 
+    if (!directory) {
+      const fnmDir =
+        this.platform === 'win32'
+          ? path.join(this.env['APPDATA'] || '', 'fnm')
```
Comment (connor4312) [inline src/targets/node/nvmResolver.ts:95]: It looks like `FNM_DIR` gets set on Windows as well, so we should use that here if set.

File: src/targets/node/nvmResolver.ts [hunk 2]
```diff
@@ -151,6 +164,15 @@ export class NvmResolver implements INvmResolver {
     return directory ? path.join(directory, 'bin') : undefined;
   }
 
+  private async resolveFnm(version: string, fnmHome: string) {
+    const directory = this.findBinFolderForVersion(
+      path.join(fnmHome, 'node-versions'),
+      `v${version}`,
+    );
+
+    return directory ? path.join(directory, 'installation', 'bin') : undefined;
```
Comment (connor4312) [inline src/targets/node/nvmResolver.ts:173]: On Windows, there's no nested `bin` directory. `node.exe` is just in `path.join(directory, 'installation')`

**Review by connor4312 — COMMENTED** [review]
A couple Windows-specific tweaks :)

File: src/targets/node/nvmResolver.ts (hunk 1 above)
Comment (thales-maciel) [reply src/targets/node/nvmResolver.ts]: thanks for the review, solved it!

File: src/targets/node/nvmResolver.ts (hunk 2 above)
Comment (thales-maciel) [reply src/targets/node/nvmResolver.ts]: thanks for the review, solved it!

**Comment by connor4312** [comment]
Thanks!

Failing tests are a result of version constraints and will be fixed with the release next week (or if I make a PR in the meantime)

**Review by connor4312 — APPROVED** [review]

**Review by TylerLeonhardt — APPROVED** [review]

**Pull request merged**